ext/mbstring/tests/koi8u_encoding.phpt

```diff
@@ -0,0 +1,26 @@
+--TEST--
+Exhaustive test of verification and conversion of KOI8-U text
+--SKIPIF--
+<?php
+extension_loaded('mbstring') or die('skip mbstring not available');
+if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
+?>
+--FILE--
+<?php
+srand(789); // Make results consistent
+include('encoding_tests.inc');
+mb_substitute_character(0x25); // '%'
+
+readConversionTable(realpath(__DIR__ . '/data/KOI8-U.txt'), $toUnicode, $fromUnicode);
+findInvalidChars($toUnicode, $invalid, $unused);
+testAllValidChars($toUnicode, 'KOI8-U', 'UTF-16BE');
+testAllInvalidChars($invalid, $toUnicode, 'KOI8-U', 'UTF-16BE', "\x00%");
+
+findInvalidChars($fromUnicode, $invalid, $truncated, map(range(0,0xFF), 2));
+testAllInvalidChars($invalid, $fromUnicode, 'UTF-16BE', 'KOI8-U', '%');
+testTruncatedChars($truncated, 'UTF-16BE', 'KOI8-U', '%');
```

Comment:
There's still a lot of repetition in these tests. It will be easier to adjust these if we reduce them down to just...

```
testSingleByteEncoding('KOI8-U', __DIR__ . '/data/KOI8-U.txt');
```

...or so.

Reply:
Indeed.